### PR TITLE
Add initial RTD configuration

### PR DESCRIPTION
**Pull request by chrysle**
* [ ] I have added an entry to `docs/changelog.md`

## Summary of changes

Related to #698 

Creates an initial ReadTheDocs configuration file. A maintainer will need to import the project on RTD using his account, and configure building for pull requests.

cc @webknjaz 

## Test plan

Tested by running
```
```

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       run: |
         nox --error-on-missing-interpreters --non-interactive --session lint
 
-  verify-docs:
+  build-docs:
```
Comment (webknjaz) [inline .github/workflows/tests.yml:36]: I don't think this should be a part of the PR.

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+  commands:
+    - mkdir --parents $READTHEDOCS_OUTPUT/html/
+    - pip install nox
+    - nox --session build_docs
+    - cp --recursive site/* $READTHEDOCS_OUTPUT/html/
```
Comment (webknjaz) [inline .readthedocs.yaml:13]: I think it's better to update `noxfile.py` to optionally accept a positional CLI argument for the target folder. This way, we could make the site generator output directly into the desired target.
```suggestion
    - nox --session build_docs -- "${READTHEDOCS_OUTPUT}"/html
```

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+version: 2
+
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+  commands:
+    - mkdir --parents $READTHEDOCS_OUTPUT/html/
```
Comment (webknjaz) [inline .readthedocs.yaml:10]: Normally, this would be created by the site generator, if you feed it with the target directory.
```suggestion
```

**Comment by webknjaz** [comment]
@chrysle looks like `publish_docs.yml` should also be deleted.

File: .readthedocs.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    - pip install nox
```
Comment (webknjaz) [inline .readthedocs.yaml:11]: I noticed that most YAML documents under `.github/` use no-space indents for sequences. Why not follow the same style? Here's a linting config I tend to use: https://github.com/cherrypy/cheroot/blob/main/.yamllint.

**Comment by webknjaz** [comment]
@gaborbernat I invited you on RTD, could you re-sync the hook @ https://readthedocs.org/dashboard/pipx/integrations/ ?

**Comment by gaborbernat** [comment]
I do not have the Github org rights for that 😊

**Comment by webknjaz** [comment]
@gaborbernat do you know who does, then?

**Comment by gaborbernat** [comment]
No idea, I'm no longer involved in this project heavily.

**Comment by webknjaz** [comment]
@gaborbernat the readme lists you first 🤷‍♂️

**Comment by gaborbernat** [comment]
Bit outdated, but most rights are with the lead maintainers, they're the ones likely that can help you with this.

**Comment by webknjaz** [comment]
@cs01 could you help us out? We're bringing in configuration for the docs migration to RTD. A maintainer with sufficient repository privileges is required to activate the integration. What's your RTD username?

**Comment by cs01** [comment]
Hi, I don’t have an RTD username. @uranusjr are you able to assist?

**Comment by webknjaz** [comment]
Invited @uranusjr on RTD. Hopefully, he has sufficient access to the repository settings too..

**Comment by cs01** [comment]
> Invited @uranusjr on RTD. Hopefully, he has sufficient access to the repository settings too..

Thanks! If he doesn’t I can give him the permissions he needs.

**Comment by chrysle** [comment]
@webknjaz Meanwhile prettified the configuration according to your suggestions! I was mostly following the guide in the RTD docs before.

File: .readthedocs.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+version: 2
+
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+  commands:
+  - pip install nox
+  - nox --session build_docs -- "${READTHEDOCS_OUTPUT}"/html
```
Comment (webknjaz) [inline .readthedocs.yaml:11]: I wonder if there's an easy way to reproduce what RTD injects by default... https://github.com/readthedocs/readthedocs.org/blob/fa54900/readthedocs/doc_builder/backends/mkdocs.py#L131-L279

**Review by webknjaz — APPROVED** [review]

**Review by dukecat0 — APPROVED** [review]

**Comment by dukecat0** [comment]
Perhaps we will need to set up a redirect from the current Github page in the future.

File: .readthedocs.yaml (hunk 5 above)
Comment (chrysle) [reply .readthedocs.yaml:11]: Do you suggest copying the code into _scripts/generate_docs.py_ with amendments?

Comment (webknjaz) [reply .readthedocs.yaml:11]: No, I wouldn't care in the context of the initial integration. Maybe, explore later.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**